lab/src/Microsoft.ServiceModel.Syndication/src/Atom10FeedFormatter.cs

```diff
@@ -961,12 +1070,18 @@ private void ReadItemFrom(XmlReader reader, SyndicationItem result, Uri feedBase
                                 {
                                     if (_preserveElementExtensions)
                                     {
-                                        CreateBufferIfRequiredAndWriteNode(ref buffer, ref extWriter, reader, _maxExtensionSize);
+                                        if (buffer == null)
+                                        {
```

Comment:
Another place where you inlined this functionality. Please put back into a method.